Sources/SwiftParserDiagnostics/SyntaxExtensions.swift

```diff
@@ -39,6 +39,16 @@ extension UnexpectedNodesSyntax {
       return nil
     }
   }
+
+  /// If this only contains one tokens satisfying `condition`, return that token, otherwise return `nil`.
+  func oneTokenSatisfying(satisfying condition: (TokenSyntax) -> Bool) -> TokenSyntax? {
```

Comment:
I thought it might be convenient to create an ad-hoc function . Let me know if there's a better way to do it.